components/image-builder-bob/pkg/proxy/auth.go

```diff
@@ -53,21 +53,30 @@ func (a MapAuthorizer) Authorize(host string) (user, pass string, err error) {
 	}()
 
 	// Strip any port from the host if present
-	host = strings.Split(host, ":")[0]
+	hostSlice := strings.Split(host, ":")
```

Comment:
This code would benefit from:
 - an explicit `parse` function
 - re-using the existing terms [hostname](https://developer.mozilla.org/en-US/docs/Web/API/Location/hostname) and [host](https://developer.mozilla.org/en-US/docs/Web/API/Location/host) not only in this code, but also all downstream functions